specs/signature-envelope-cose.md

```diff
@@ -239,6 +239,52 @@ The final signature envelope is a `COSE_Sign1_Tagged` object, consisting of Payl
 )
 ```
 
+## COSE Hash Envelope
+Notary Project also supports [COSE Hash Envelope](https://datatracker.ietf.org/doc/draft-ietf-cose-hash-envelope/05/). COSE Hash Envelope is essentially another format of [COSE Sign1 signature structure](https://datatracker.ietf.org/doc/html/rfc8152#section-4.2). 
+The differences between a `COSE Hash Envelope` and the signature envelope defined under [Signature Envelope](./signature-envelope-cose.md/#signature-envelope) are as follows:
+1. Instead of `targetArtifact` as shown under [COSE Payload](./signature-envelope-cose.md/#cose-payload), the payload of a `COSE Hash Envelope` is the hash of the original content bytes that is being signed. The hash algorithm MUST be deduced from the signing certificate's public key (see [Algorithm Selection](./signature-specification.md#algorithm-selection)).
```

Comment:
COSE Hash Envelope does not use `targetArtifact` defined in `COSE Payload` as the payload or pre-image payload. Instead, it decomposes the information specified in `argetArtifact` to COSE Hash Envelope specific headers. Basically, there is a mapping:
- `targetArtifact.mediaType` --> `cose_sign1.protected_header.preimage_content_type`
- `targetArtifact.digest` --> `cose_sign1.payload`

Note that `targetArtifact.size`, `targetArtifact,annotations`, and `targetArtifact.artifactType` are lost. We may think about other ways to support them.